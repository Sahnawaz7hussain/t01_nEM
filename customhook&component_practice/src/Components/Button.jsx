//Complete the custom Input component based on the props that it expects to receive
import styles from "./button.module.css";
export const Button = ({
  colorScheme = "white",
  size = "md",
  variant,
  onClick,
  borderRadius = "5px",
  children,
}) => {
  return (
    <button
      style={{
        borderRadius: `${borderRadius}`,
      }}
      className={`${styles[colorScheme]} ${styles[size]} ${styles[variant]} `}
      onClick={onClick}
    >
      {children}
    </button>
  );
};
