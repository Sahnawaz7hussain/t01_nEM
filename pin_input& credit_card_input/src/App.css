.App {
  text-align: center;
}
.pinInput {
  height: 40px;
  width: 40px;
  font-size: 22px;
  margin: 16px 4px;
  text-align: center;
}
.creditCard {
  height: 220px;
  width: 380px;
  border: 1px solid blue;
  border-radius: 5px;
  margin: auto;
  background-color: lightcyan;
  box-sizing: border-box;
  padding-top: 50px;
}
.cardInput {
  height: 40px;
  width: 60px;
  font-size: 20px;
  margin: 16px 4px;
  text-align: center;
}
