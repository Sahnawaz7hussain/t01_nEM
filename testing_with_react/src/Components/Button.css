.btn {
  color: white;
  cursor: pointer;
  font-size: 30px;
  margin: 3rem 0;
}
.red {
  background-color: red;
}
.green {
  background-color: green;
}
.yello {
  background-color: yellow;
}
.orange {
  background-color: orange;
}
.tomato {
  background-color: tomato;
}
.ghost {
  border: none;
}
.bordered {
  border: 3px solid black;
}
