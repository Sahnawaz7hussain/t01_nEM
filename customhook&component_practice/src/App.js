import "./App.css";
import { Button } from "./Components/Button";
import { Input } from "./Components/Input";
import { useCounter } from "./hooks/useCounter";

function App() {
  const { countValue, incCount, decCount } = useCounter({
    initialValue: 10,
    minValue: 5,
    maxValue: 15,
  });

  return (
    <div className="App">
      {/* DO NOT CHANGE the 5 lines below   */}
      <h3>Value: {countValue}</h3>
      <button onClick={() => incCount()}>Increment</button>
      <button onClick={() => incCount(3)}>Increment 3</button>
      <button onClick={() => decCount(2)}>Decrement 2</button>
      <button onClick={() => decCount(4)}>Decrement 4</button>
      <button onClick={() => decCount()}>Decrement 2</button>

      <br />
      <br />
      <Button colorScheme="orange" size="xl" variant="solid">
        Button1
      </Button>
      <Button colorScheme="green" size="lg" variant="solid">
        Button2
      </Button>
      <Button colorScheme="red" size="md" variant="solid">
        Button3
      </Button>
      <Button colorScheme="yellow" size="sm" variant="solid">
        Button4
      </Button>
      {/* <Button colorScheme="orange" size="xl" variant="solid" />
      <Button colorScheme="green" size="md" variant="ghost" />
      <Button colorScheme="red" size="sm" variant="solid" /> */}
      <br />
      <Input size="xl" variant={"outline"} />
    </div>
  );
}

export default App;
