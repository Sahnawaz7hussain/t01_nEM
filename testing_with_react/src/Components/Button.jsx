import React from "react";
import "./Button.css";

const Buttons = ({ children, colorScheme, variant, onClick }) => {
  return (
    <button
      data-testid="customButton"
      onClick={onClick}
      className={`btn ${colorScheme || "red"} ${variant || "ghost"} `}
    >
      {children}
    </button>
  );
};

export default Buttons;
