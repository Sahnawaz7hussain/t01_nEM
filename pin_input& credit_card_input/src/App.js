import { useState } from "react";
import "./App.css";
import CreditCard from "./Components/creditCard/CreditCard";
import Pin from "./Components/Pin";

function App() {
  const [pinInput, setPinInput] = useState("");
  return (
    <div className="App">
      <Pin
        className="inputBox"
        length={5}
        maxLength={1}
        setPinInput={setPinInput}
      />
      <h2>OTP is: {pinInput}</h2>
      <br />
      <CreditCard length={4} maxLength={4} />
    </div>
  );
}

export default App;
