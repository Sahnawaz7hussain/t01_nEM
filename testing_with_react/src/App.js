import { useState } from "react";
import "./App.css";
import Buttons from "./Components/Button";

function App() {
  const [theme, setTheme] = useState("Light");
  return (
    <div className="App">
      <h3>Current Theme is {theme} </h3>
      <Buttons
        colorScheme={"tomato"}
        variant={"bordered"}
        onClick={() => setTheme(theme === "Dark" ? "Light" : "Dark")}
      >
        click me
      </Buttons>
    </div>
  );
}

export default App;
