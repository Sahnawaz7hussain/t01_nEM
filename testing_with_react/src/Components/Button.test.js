import { render, screen } from "@testing-library/react";
import App from "../App";
import Button from "./Button";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

describe("Testing the Custom component", () => {
  it("should be in the DOM tree", () => {
    render(<Button>Testing Button</Button>); // virtually.
    //screen.debug();

    let button = screen.getByText("Testing Button");
    // document.get
    expect(button).toBeInTheDocument();
  });

  it("should have a button with the text 'Click Me' in the App.js", () => {
    render(<App />);
    // screen.debug();
    let button = screen.getByTestId("customButton");
    expect(button).toBeInTheDocument();
  });
  it("should check if the label is present as passed throught the props", () => {
    render(<Button>Testing</Button>);
    let button = screen.getByTestId("customButton");
    expect(button).toHaveTextContent("Testing");
  });
  it("should be an empty DOM element, if the children props is not passed", () => {
    render(<Button></Button>);
    let button = screen.getByTestId("customButton");
    expect(button).toBeEmptyDOMElement();
  });
  it("should the theme when the button button is clicked", () => {
    render(<App />);
    let button = screen.getByTestId("customButton");

    let h3Tag = screen.getByText("Current Theme is Light");
    expect(h3Tag).toHaveTextContent("Light");

    userEvent.click(button);
    expect(h3Tag).toHaveTextContent("Dark");
  });

  it("should create/match the snapshot", () => {
    const tree = renderer
      .create(
        <Button colorScheme={"green"} variant={"bordered"}>
          Custom Button
        </Button>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
