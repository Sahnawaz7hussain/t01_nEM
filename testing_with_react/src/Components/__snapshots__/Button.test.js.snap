// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Custom component should create/match the snapshot 1`] = `
<button
  className="btn green bordered "
  data-testid="customButton"
>
  Custom Button
</button>
`;
